build/terraform/e2e/module.tf

```diff
@@ -62,22 +62,50 @@ variable "kubernetes_versions" {
   }
 }
 
+variable "test_names" {
+  description = "Use the same terraform templates for both e2e and upgrade tests. Includes test name and initial node counts for standard clusters."
+  type        = map(number)
+  default     = {
+    "e2e" = 10
+    "upgrade" = 4
+  }
+}
+
+// Handle nested loop in terraform. Flatten combines the two maps into a list.
+locals {
```

Comment:
So now the creation of e2e test clusters and upgrade test clusters are coupled in this module file,  which sounds not convenient. For example, `make gcloud-e2e-test-cluster` will create both set of clusters. Can we move the creation of upgrade test clusters to a separate module file, which still use the same `gke-autopilot/module.tf` and `gke-standard/module.tf`? The structure in my mind is something like below, WDYT?
- build/terraform/test-clusters
  - e2e
    - module.tf
  - upgrade
    - module.tf
  - gke-standard
    - module.tf
  - gke-autopilot
    - module.tf
  - state-bucket
    - main.tf

Reply:
Good point, I didn't think about people using this for local e2e creation. I'll break it out into separate modules.